### ISSUE TITLE
html/semantics/forms/the-input-element/datetime-local.html incorrectly expects underflow/overflow sanitization

### ISSUE THREAD

**Issue by pshaughn** [web-platform-tests/wpt#20951]
https://html.spec.whatwg.org/multipage/input.html#range-state-(type=range) explicitly says that, for a type="range" input, min and max are enforced at sanitization time when suffering from underflow/overflow. _This is the only input type specified as doing so._ By contrast, https://html.spec.whatwg.org/multipage/input.html#local-date-and-time-state-(type=datetime-local) does not say to enforce underflow and overflow when sanitizing, and therefore it is legal for a datetime-local input to hold an out-of-range value; such a value makes the element "invalid", but that's a separate step from sanitization.

The two tests in https://wpt.fyi/results/html/semantics/forms/the-input-element/datetime-local.html?label=experimental&label=master&aligned that have a full cross-browser failure have an incorrect expectation, as they set up an out-of-range but otherwise legal value and expect it to have overflow/underflow sanitization applied to it.

**Comment by guest271314** [comment]
> that have a full cross-browser failure

Nightly 73 fails for only the two subtests 

- Value < min attribute
- Value > max attribute

Since only the last four elements of `datetimeLocal` have `attributes` property the test could use `if` statement 

      if (w.attributes) {
        for(var attr in w.attributes) {
          input[attr] = w.attributes[attr];
        }
      }

Why is `"T"` expected in output of `input.value`?

**Comment by pshaughn** [comment]
Good point about the "T", every browser happens to be using "T" but the combination of https://html.spec.whatwg.org/multipage/input.html#local-date-and-time-state-(type=datetime-local):value-sanitization-algorithm and https://html.spec.whatwg.org/multipage/common-microsyntaxes.html#valid-local-date-and-time-string implies that it could actually be either a T or a space.

**Comment by guest271314** [comment]
>  every browser happens to be using "T" 

Did not get a `"T"` for output of `.value` at Chromium 80 or Firefox 71 at *nix 32-bit when tested earlier.

**Comment by pshaughn** [comment]
That surprises me, as I wouldn't have expected that depends on the OS, but it sounds like it does and the tests definitely need fixing.

**Comment by guest271314** [comment]
For 

`{value: "2014-01-01 11:11:11.111"
  , expected: "2014-01-01T11:11:11.111"
  , testname: "datetime-local input value set to 2014-01-01 11:11:11.111 without min/max"
}`

why is a `"T"` expected in output?

**Comment by pshaughn** [comment]
By my reading of the spec, that's a problem with the test and it actually should be accepting either a T or a space! The specific browser versions in continuous integration happen to return a T, so this wasn't caught before now.

**Comment by guest271314** [comment]
Deleted previous comment as the statement was not based on thorough testing, over and over.  Have not tried Nightly 73, yet. Did a cursory set of tests earlier using the code from the current test.

Does this assertion fail at the browsers you are tried the code?

```
var input = document.createElement("input");
i.type = "datetime-local";
i.value = "2014-01-01 11:11:11.111";
console.assert(i.value.includes("T"), [i.value]);
```

**Comment by pshaughn** [comment]
There's definitely a problem here, @guest271314, but it's a different problem from the min/max one so I've filed another issue for it.

**Comment by guest271314** [comment]
The assertion within
```
var input = document.createElement("input");
input.type = "datetime-local";
input.value = "2014-01-01 11:11:11.111";
console.assert(input.value.includes("T"), [input.value]);
```

fails at Nightly 73.

Results:

- Chromium 80: `""` (empty string)
- Nightly 73: `"2014-01-01 11:11:11.111"` (input string)

-- 

(One issue with using continuous builds is ascertaining precisely which preferences or flags are set in order to determine if the problem is the test, the browser, or both (https://github.com/web-platform-tests/wpt.fyi/issues/1648).)

**Comment by pshaughn** [comment]
@guest271314: It turns out the T versus space question is specified after all as requiring the T to be present after sanitization. https://html.spec.whatwg.org/multipage/input.html#local-date-and-time-state-(type=datetime-local):value-sanitization-algorithm refers to https://html.spec.whatwg.org/multipage/common-microsyntaxes.html#valid-normalised-local-date-and-time-string which requires the T specifically, and also requires the shortest possible string.

**Comment by guest271314** [comment]
What is the expected result for 
```
input.type = "datetime-local";
input.value = "2014-01-01 11:11:11.111";
```
?

Is a `"T"` expected in the output?

Do the values output by Chromium and Nightly at https://github.com/web-platform-tests/wpt/issues/20951#issuecomment-570095312 meet the expectation?
